src/lib/y2network/proposal_settings.rb

```diff
@@ -123,12 +123,14 @@ def virtual_proposal_required?
     # depending on the backend selected during the proposal and the packages
     # installed
     def network_service
-      case current_backend
-      when :network_manager
-        network_manager_installed? ? :network_manager : :wicked
-      else
-        :wicked
+      if current_backend == :network_manager
+        return :network_manager if network_manager_installed?
+
+        log.info("NetworkManager is the selected service but it is not installed")
```

Comment:
Why not use just a single line?

Reply:
Probably will this is enough, I tried to move the ones from the network_service method in the NetworkAutoYast class but in fact this is already enough and were everything is checked.